src/predicate.rs

```diff
@@ -185,6 +309,63 @@ impl<A: Predicate, B: Predicate> Predicate for Descendant<A, B> {
                 node = parent;
             }
         }
-        return false;
+        false
+    }
+}
+
+/// Matches if inner Predicate `B` matches the node and `A` matches the node before this one.
+///
+/// This is equivalent to the `before + after` CSS selector.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct ImmediatelyAfter<A, B>(pub A, pub B);
+
+impl<A: Predicate, B: Predicate> Predicate for ImmediatelyAfter<A, B> {
+    fn matches(&self, node: &Node) -> bool {
+        self.1.matches(node) && node.prev().map_or(false, |prev| self.0.matches(&prev))
+    }
+}
+
+/// Matches if inner Predicate `B` matches the node and `A` matches any node before this one.
+///
+/// This is equivalent to the `before ~ after` CSS selector.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct After<A, B>(pub A, pub B);
```

Comment:
The CSS selector is officially called the General Sibling Combinator. What do you think about naming this `Sibling`?

Reply:
Also will do